e2e/files.spec.ts

```diff
@@ -0,0 +1,283 @@
+      await deleteFileHandleWithRetry(accessToken, fileHandleId, page)
+    }
+
+    await context.close()
+  })
+
+  testAuth(
+    'File sharing',
+    async ({ userPage, validatedUserPage, browserName }, testInfo) => {
+      if (browserName === 'webkit') {
+        test.info().annotations.push({
+          type: 'very slow',
+          description: `webkit in CI only. May be related to the InviteWidget. 
+            Evaluate if timeout can be removed after addressing 
+            https://sagebionetworks.jira.com/browse/SWC-6569.`,
+        })
+        test.setTimeout(testInfo.timeout * 5)
+      }
+
+      const userName = userConfigs['swc-e2e-user'].username
+      const validatedUserName = userConfigs['swc-e2e-user-validated'].username
+
+      const { fileEntityId, fileName } = await testAuth.step(
+        'Create a file',
+        async () => {
+          const userAccessToken = await getAccessTokenFromCookie(userPage)
+          const fileName = `${generateEntityName('file')}.txt`
+          const { fileEntityId, fileHandleId } = await createFile(
+            fileName,
+            'Created for SWC e2e testing.',
+            'text/txt',
+            userProject.id,
+            userAccessToken,
+            userPage,
+          )
+          fileHandleIds.push(fileHandleId)
+          return { fileEntityId, fileName }
+        },
+      )
+
+      await testAuth.step('First user can view a private file', async () => {
+        await userPage.goto(`${entityUrlPathname(userProject.id)}/files`)
+        await waitForInitialPageLoad(userPage)
+
+        const fileLink = userPage.getByRole('link', { name: fileName })
+        await expect(fileLink).toBeVisible()
+
+        await fileLink.click()
+        await expect(
+          userPage.getByText(`Discussion about ${fileName}`),
+        ).toBeVisible()
+      })
+
+      await openFileSharingSettings(userPage)
+      await confirmSharingSettings(userPage, userName, validatedUserName, false)
+
+      await testAuth.step('Second user cannot access the file', async () => {
+        await validatedUserPage.goto(entityUrlPathname(fileEntityId))
+        await waitForInitialPageLoad(validatedUserPage)
+
+        await expectNoAccessPage(validatedUserPage)
+      })
+
+      await testAuth.step(
+        'First user enables Local Sharing Settings for the File',
+        async () => {
+          await userPage
+            .getByRole('button', { name: 'Create Local Sharing Settings' })
+            .click()
+          await expect(
+            userPage.getByRole('heading', { name: 'Add More People' }),
+          ).toBeVisible()
+        },
+      )
+
+      await testAuth.step(
+        'First user shares the file with second user',
+        async () => {
+          const userSuggestBox = userPage.getByRole('textbox', {
+            name: 'Enter name...',
+          })
+          await userSuggestBox.fill(validatedUserName)
+          await userSuggestBox.press('Shift')
```

Comment:
`press` and `pressSequentially` are particularly slow in webkit in CI, so fill the validated username to avoid calling `press` for each letter of the  validated user's name, then `press` Shift to trigger the user suggestion dropdown, as discussed in [SWC-6569](https://sagebionetworks.jira.com/browse/SWC-6569)